models/mit_ipea_project/vw_ticketing.sql

```diff
@@ -0,0 +1,40 @@
+-- ticketing view, ticketing data cleaned for:
+-- - card types that do not unique identify users (these include cash transactions)
+-- - pre-identified card ids with extreme values
+
+WITH raw_ticketing AS (SELECT data                                                               AS as_at,
+                              EXTRACT(time from datetime)                                        AS origin_time,
+                              id_cartao                                                          AS card_id,
+                              tipo_cartao                                                        AS card_type,
+                              ROW_NUMBER() OVER (PARTITION BY id_cartao, data ORDER BY datetime) AS daily_trip_id,
+                              id_empresa                                                         AS company_id,
+                              id_veiculo                                                         AS vehicle_id
+
+                       FROM `rj-smtr.br_rj_riodejaneiro_onibus_tg.transacao`
+                       WHERE tipo_cartao NOT IN (SELECT chave
+                                                 FROM `rj-smtr.br_rj_riodejaneiro_onibus_tg.dicionario`
+                                                 -- Card types to drop:
+                                                 WHERE VALOR IN (
+                                                                 'RIOONIBUS - BOTOEIRA',
+                                                                 'RIOONIBUS - ROD. CIDADAO RIO ONIBUS',
+                                                                 'RIOONIBUS - RODOVIARIO RIO ONIBUS'
+                                                     )))
+
+SELECT *,
+       CASE
+           WHEN MAX(daily_trip_id) OVER (PARTITION BY card_id, as_at) = 1
+               THEN 'Only transaction'
+           WHEN MAX(daily_trip_id) OVER (PARTITION BY card_id, as_at) = daily_trip_id
+               THEN 'Last transaction'
+           WHEN MIN(daily_trip_id) OVER (PARTITION BY card_id, as_at) = daily_trip_id
+               THEN 'First transaction'
+           ELSE
+               'Intermediate transaction'
+           END
+           AS daily_trip_stage
+
+FROM raw_ticketing
+# These card_ids have an average of >400 transactions per day
```

Comment:
Is this `#` comment working?

Reply:
Works on my end but, agree, it is unusual to use in SQL. Updated to `--`.